### ISSUE TITLE
Quick question about SPM format & ModelMatch

### ISSUE THREAD

**Issue by martinrehfeld** [mikejr83/SPMReader#1]
As you analysed the SPM format extensively you will probably know...

Is the part that ModelMatch adds to the transmitter GUID part of the SPM?

(I would like to use two model slots with the same receiver and was hoping to be able to achieve that by manually tweaking the SPM file)

Thank you & cheers,
Martin

**Comment by mikejr83** [comment]
On the DX18 the only thing I could think of that would be of help would be this line: 

> ; Originator="HH208JABAzAIUNETsEyzf8WKkPVdMf; mmNum=0; bCode=3"

I'm not sure if that will give you what you want. I don't know what `mmNum` is, but it could be an abbreviation for _Model Match Number_. 

It seems to vary in the SPM files that I exported from my radio. In OpenTX the concepts for model and RX are two separate identities. You can bind two different model entries to the same RX number. Since a lot of this functionality is very similar between all radios (except, probably Futaba) I'd be willing to be that is what is going on in this case. 

Try this:
Create one model, bind it, and set it up for a model. Then export it and see what you have for the "originator" line, specifically what value _mmNum_ is set. Create a second model (in the radio) and set it up exactly like you did the first. Export the SPM file and see if _mmNum_ is incremented by 1. I bet it will be.

Try changing the _mmNum_ values to match and then reimporting that second SPM file. A few different things could happen. The best case is the file imports and you now have three models, the first one, the second one, and a third newly created model from the import. Models one and three would be different in their settings but _bound_ to the same receiver allowing either one to be selected without having to rebind. 

However, Spektrum/HH could have created some kind of error checking. Upon import the radio could verify that no two models share a _mmNum_ value and therefore either increment it to an unused value or throw some sort of error. The error could be that it won't import, that the model won't respond, or you couldn't load the model.

Also, everything could seem to import fine, but the model won't work. I ran into that when editing by hand.

**WHATEVER YOU DO** do not test this with an electric motor connected. Also, please, please make sure everything works on the ground!

Please update with your results.

**Comment by martinrehfeld** [comment]
Thank you for sharing your knowledge! Here are the results of the experiment :-)

tl;dr **I found no way to have two model slots bound to the same RX**

After copying the model in the TX and exporting it, the resulting SPM files differed in only 3 fields:

```
mmNum # went from 2 -> 7
bCode # went from 4 -> 0
Name  # obviously
```

I changed `mmNum` back to 2 and re-imported the file. The resulting new model does NOT bind to the same RX. After exporting that, I found that the TX assigned `mmNum=8` to it. So obviously there is a safeguard in place that keeps `mmNum` unique on imports.

I was still curious if the `bCode` would play a role in this, so I created a file that has `mmNum=2` and `bCode=4` just as the original model. After importing the TX assigned a new `mmNum=9` and also reset `bCode=0`. Even importing the original bound model updates the `mmNum=9`, but leaves `bCode=4` unchanged.

My conclusion is that `mmNum` is probably the field that determines the ModelMatch GUID, but there is no way to get two models to have the same `mmNum` by means of using the SD Card import utility.

Cheers,
Martin

**Comment by mikejr83** [comment]
Dang! Good to know at least. Thanks for testing those theories. You might want to look up AndyKuntz on RCGroups. He's a Spektrum dev and will sometimes answer questions. You might ask him if there could be a future update of the AirWave software that would allow this.
